### PR TITLE
Untangle: Update Social Connections Link

### PR DESCRIPTION
**Pull request by DustyReagan**
Fixes https://github.com/Automattic/dotcom-forge/issues/5552

## Proposed changes:

* This PR updates the Jetpack Social link & button to redirect to the Jetpack Cloud Social Connections page for sites with access to Jetpack Cloud. (Note that proxied Atomic sites with the Classic admin interface enabled are now granted access to Jetpack Cloud via D138559-code)
* This PR also updates the link copy from "WordPress.com sharing page" to "Connections page". This way the copy makes sense whether it redirects to Calypso or Jetpack Cloud.

Atomic Site WP-Admin
Before | After
--|--
classic-before | classic-after

Jetpack Site
Before | After
--|--
jetpack-before  | jetpack-after

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1708126615252529-slack-CDLH4C1UZ

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Use the Jetpack Beta plugin to load this PR.
* On a Jetpack site goto /wp-admin/options-general.php?page=sharing click on the "Connections page" link and "Jetpack Social Settings" button highlighted in the picture above. The links should redirect you to the Jetpack Cloud "Social Connections" page at https://cloud.jetpack.com/jetpack-social/[site-id]
* The above should also be true on an Atomic site with Classic admin enabled while you're proxied.
* Test the links above un-proxied. They should now redirect you to Calypso https://wordpress.com/marketing/connections/[site-id]?site=[site-id]
* Now test proxied, but with "Default" admin view enabled. It should redirect you to Calypso https://wordpress.com/marketing/connections/[site-id]?site=[site-id]

### CODE REVIEW

**Comment by kraftbj** [comment]
First change to the Publicize package since the last release so added the changelog version bumps so CI is happy once reviewed.

File: projects/packages/publicize/src/class-publicize-ui.php [hunk 1]
```diff
@@ -133,12 +133,12 @@ public function admin_page() {
 						),
 					)
 				),
-				esc_url( $this->publicize->publicize_connections_url() )
+				esc_url( $this->publicize->publicize_connections_url( 'jetpack-social-connections-classic-editor' ) )
```
Comment (lsl) [inline projects/packages/publicize/src/class-publicize-ui.php:136]: I think this needs to be limited to classic view + proxy users?

@ebinnion Should we be sending non classic view atomic users to Jetpack cloud once we release this? (release = remove proxy requirement, but not the classic view requirement)

Comment (DustyReagan) [reply projects/packages/publicize/src/class-publicize-ui.php:136]: > I think this needs to be limited to classic view + proxy users?

I believe it is due to D138559-code

**Comment by ebinnion** [comment]
Your PR as is tests well. But, should we also handle this connections link in the Jetpack > Settings > Sharing page?

Screenshot 2024-02-19 at 12 51 19 PM

File: projects/packages/publicize/src/class-publicize-ui.php (hunk 1 above)
Comment (lsl) [reply projects/packages/publicize/src/class-publicize-ui.php:136]: A few things came out of this conversation p1708126615252529-slack-CDLH4C1UZ

I'm pretty sure this is what we need:

Atomic + default => calypso
Atomic + classic => calypso
Atomic + classic + proxy => jetpack cloud
Atomic + default + proxy => calypso
Jetpack self hosted => calypso
Jetpack self hosted + proxy => calypso
Jetpack social only self hosted => jetpack cloud

I don't think D138559-code is doing this yet.

Atomic + classic => calypso - We'll need to make sure this switches to Jetpack cloud on release. 

This all assumes that we can use jp cloud for social connections on atomic,

Atomic + classic + proxy => jetpack cloud

We need to test this works to make a connection in calypso, toggle to classic view, see the same connection on jetpack cloud. If that doesn't work we'll need to implement connections in site tools.

Comment (pablinos) [reply projects/packages/publicize/src/class-publicize-ui.php:136]: > We need to test this works to make a connection in calypso, toggle to classic view, see the same connection on jetpack cloud. If that doesn't work we'll need to implement connections in site tools.

You should see all the same connections whether you're in Calypso or Cloud, it's the same components and API calls.

**Comment by lsl** [comment]
See https://github.com/Automattic/dotcom-forge/issues/5552#issuecomment-1953529953

p1708342763524749/1708339800.549179-slack-C06DN6QQVAQ

No longer need to fix this as we're removing the menu item instead https://github.com/Automattic/dotcom-forge/issues/5672

**Pull request closed without merging**